client/src/core/BaseTypes.res

```diff
@@ -1,11 +1,16 @@
 /// Basic types
+/// Does this *extend* basic types, or what? `pp` is a reall unclear fn name.
 
 module Belt = {
+  // what is this? Could it somehow be more clear?
   include (Belt: module type of Belt with module Map := Belt.Map and module Result := Belt.Result)
 
   module Result = {
     include Belt.Result
 
+    // what does this do?
+    // When I remove it, it errors in RuntimeTypes.res, but I can't figure out why/where.
+    // something related to all of these
     let pp = (
       okValueFormatter: (Format.formatter, 'okValue) => unit,
       errValueFormatter: (Format.formatter, 'errValue) => unit,
```

Comment:
this is a recurring mystery for me. I've found some notes online about `@ppx` but I don't understand that either, yet.

Reply:
The deriving ppx requires parent types to have a formatting function, typically named `pp` or `pp_typename`. Built-in types do not have these annotations, so we need to add them.

Ideally we'd have a better way to pretty print structures, since PPX deriving won't work on future versions of rescript. But until we remove it, nearly all types need to support this.

Reply:
The specific stuff here is how to add functionality to existing modules. You sort of redefine them pulling in the old functionality, but sometimes you want to redefine some module inside a module, and that's when things get funky.